### DISCUSSION TITLE
Questions component

### DISCUSSION THREAD

**Discussion by tobisupreme** [AltCamp/altcampv1-backend#76] [Ideas]
Documenting the process of implementing "Questions"

**Comment by tobisupreme** [comment]
from our sketch on the [miro board](https://miro.com/app/board/uXjVPzTIF5I=/), our `Question` schema can be represented as

```javascript
{
  title: {
    type: String,
    required: true,
  },
  body: {
    type: String,
    required: true,
  },
  upvotes: {
    type: Number,
    default: 0,
  },
  downvotes: {
    type: Number,
    default: 0,
  },
  author: {
    type: ObjectId,
    ref: 'Account',
    required: true,
  },
  answer: {
    type: [ObjectId],
    ref: 'Answer',
  },
};
```

**Reply by jhhornn** [reply]
I have adopted storing just the ID of the user that either upvoted or downvoted like you sent. So that way we can keep track of the number of element in the array for their respective totals

**Reply by tobisupreme** [reply]
Nice one!

I introduce the changes to the questions schema in 6638b6a

**Reply by tobisupreme** [reply]
> I have adopted storing just the ID of the user that either upvoted or downvoted like you sent. So that way we can keep track of the number of element in the array for their respective totals

I have both

**Comment by tobisupreme** [comment]
commit [f0e55a1](https://github.com/StudyBuddyv1/studybuddyv1-backend/commit/f0e55a14c0659360de9b34d808b90495cb4b6801) introduces a `Question` schema and three (3) blank scripts

![image](https://user-images.githubusercontent.com/98078707/221428594-52a2cdf7-f586-43a7-8ee6-cfb401e9ea4c.png)

**Comment by tobisupreme** [comment]
Even though it is _not very pleasant to write tests before the code_, adopting TDD yields advantages, especially for/on long-term bases. In commit 68c3bb3, we add some tests.

**Comment by iamaamunir** [comment]
@tobisupreme @Team. I don't think its ideal to have `min()` validator set for both the title and body in the question validation. Think it through.
